NameGame/NameGame/UI/Game/GameViewController.swift

```diff
@@ -0,0 +1,164 @@
+    
+    class func make() -> GameViewController {
+        return UIStoryboard.main.make()
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        lifecycleObserver?.viewDidLoad()
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        self.navigationController?.navigationBar.isHidden = true
+        
+        lifecycleObserver?.viewWillAppear(animated)
+    }
+    
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        lifecycleObserver?.viewDidAppear(animated)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        lifecycleObserver?.viewWillDisappear(animated)
+    }
+    
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        lifecycleObserver?.viewDidDisappear(animated)
+    }
+    
+    @IBAction func userButtonPressed(_ sender: UIButton) {
+        guard let viewModel = self.viewModel else {
+            return
+        }
+        
+        let index = sender.tag
+        if index >= 0, index < viewModel.choices.count {
+            currentSelectedChoiceIndex = index
+            
+            let selectedUser = viewModel.choices[index]
+            eventHandler.didGetUserChoice(selectedUser)
+        }
+    }
+    
+    @IBAction func endGameButtonPressed(_ sender: Any) {
+        eventHandler.didPressEndGame()
+    }
+    
+    func inject(_ eventHandler: GameViewEventHandler) {
+        self.eventHandler = eventHandler
+    }
+
+}
+
+extension GameViewController: StoreSubscriber {
+    
+    func newState(state: GameViewModel?) {
+        viewModel = state
+        guard let gameViewModel = viewModel else {
+            return
+        }
+        
+        renderImageDictionary(with: gameViewModel)
+        renderChoices(with: gameViewModel)
+        renderResult(with: gameViewModel)
+        renderScore(with: gameViewModel)
+        renderEndGame(with: gameViewModel)
+    }
+    
+    func renderImageDictionary(with viewModel: GameViewModel) {
+        self.imageDictionary = viewModel.imageDictionary
+    }
+    
+    func renderChoices(with viewModel: GameViewModel) {
+        userButtons[currentSelectedChoiceIndex].layer.borderWidth = 0.0
+        userButtons[currentCorrectChoiceIndex].layer.borderWidth = 0.0
+        
+        for (index, choice) in viewModel.choices.enumerated() {
+            let image = imageDictionary[choice]
+            
+            let button = userButtons[index]
+            button.setImage(image, for: .normal)
+            button.tag = index
+            
+            if choice == viewModel.correctChoice {
+                currentCorrectChoiceIndex = index
+            }
+        }
+        
+        nameLabel.text = viewModel.correctChoice.firstName + " " + viewModel.correctChoice.lastName
```

Comment:
Extra credit; check out this tool instead of string concatenation. https://developer.apple.com/documentation/foundation/personnamecomponentsformatter